Sources/Build/BuildOperationBuildSystemDelegateHandler.swift

```diff
@@ -214,34 +211,53 @@ final class TestEntryPointCommand: CustomLLBuildCommand, TestBuildCommand {
             throw InternalError("main file output (\(TestEntryPointTool.mainFileName)) not found")
         }
 
-        let testObservabilitySetup: String
-        if self.context.buildParameters.testingParameters.experimentalTestOutput
-            && self.context.buildParameters.targetTriple.supportsTestSummary {
-            testObservabilitySetup = "_ = SwiftPMXCTestObserver()\n"
-        } else {
-            testObservabilitySetup = ""
-        }
-
         // Write the main file.
         let stream = try LocalFileOutputByteStream(mainFile)
 
-        stream.send(
-            #"""
-            \#(generateTestObservationCode(buildParameters: self.context.buildParameters))
+        if self.context.buildParameters.testingParameters.library == .swiftTesting {
+            stream.send(
+                #"""
+                #if canImport(Testing)
+                @_spi(SwiftPackageManagerSupport) import Testing
+                @main struct Runner {
+                    static func main() async {
+                        await Testing.swiftPMEntryPoint() as Never
+                    }
+                }
+                #endif
+                """#
+            )
+        } else {
```

Comment:
probably better to switch over both cases rather than if/else which assumes the seconds is always xctest